src/data.ts

```diff
@@ -0,0 +1,31 @@
+import { LocData } from './types.d.ts';
+
+/** fetchWrapper returns an instance of fetch wrapped for use with a single resource with def data and error handling */
+const fetchWrapper = (url: string, dataHandler?: any, errorHandler?: any) => {
+  return async () => {
+    try {
+      const res = await fetch(url);
+      const data = await res.json();
+      return dataHandler ? dataHandler(data) : data;
+    } catch (err) {
+      return errorHandler ? errorHandler(err) : console.log(err);
+    }
+  };
+};
```

Comment:
I think it would be beneficial to test each function independently. Here are some tests that I'd ideally like to see:

```js
// happy path with url only
Deno.test('fetchWrapper returns expected data with default handler');

// happy path with custom success handler
Deno.test('fetchWrapper returns expected data with custom handler');

// error path with url only
Deno.test('returns expected error response from default handler');

// error path with custom handler
Deno.test('returns expected error response from custom handler');
```

Reply:
Worth noting that as I stubbed out those tests, it occurred to me that there might be a case where you only want to pass a custom error handler, but don't need a custom success handler. It might be worth refactoring so `fetchWrapper` takes an object as an argument.

```js
// you'd need to include types, but this is the basic idea
const fetchWrapper = ({
  url,
  dataHandler,
  errorHandler,
}) => {...};
```